extensions/azure/processors/AzureBlobStorageSingleBlobProcessorBase.h

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file AzureBlobStorageSingleBlobProcessorBase.h
+ * AzureBlobStorageSingleBlobProcessorBase class declaration
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <utility>
+
+#include "AzureBlobStorageProcessorBase.h"
+
+namespace org::apache::nifi::minifi::azure::processors {
+
+class AzureBlobStorageSingleBlobProcessorBase : public AzureBlobStorageProcessorBase {
+ public:
+  // Supported Properties
+  EXTENSIONAPI static const core::Property Blob;
```

Comment:
I don't think it's that complex at the moment as for a blob operation we have 4 and for a listing processor we have 3 levels of hierarchy:
`PutAzureBlobStorage->AzureBlobStorageSingleBlobProcessorBase->AzureBlobStorageProcessorBase->AzureStorageProcessorBase`
`ListAzureBlobStorage->AzureBlobStorageProcessorBase->AzureStorageProcessorBase`

I think to see the differences in the properties in each inheritance level is clearer than if we would add composition into the mix. If we would for example compose the AzureBlobStorageSingleBlobProcessorBase level it would be harder to see the differences between the listing and the blob operations. 

Also this is the last Azure processor to be implemented, so the hierarchy should not increase (if it does then I also think we might have to think over). But I would be curious what would you specifically suggest to be part of a composition, maybe something I didn't think of.